velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -960,37 +960,40 @@ struct SecondsToTimestampFunction {
   VELOX_DEFINE_FUNCTION_TYPES(TExec);
 
   template <typename T>
-  FOLLY_ALWAYS_INLINE void call(out_type<Timestamp>& result, T seconds) {
+  FOLLY_ALWAYS_INLINE bool call(out_type<Timestamp>& result, T seconds) {
     if constexpr (std::is_integral_v<T>) {
       result = Timestamp(static_cast<int64_t>(seconds), 0);
+      return true;
     } else {
+      if (FOLLY_UNLIKELY(std::isnan(seconds) || std::isinf(seconds))) {
+        return false;
+      }
+
       // Cast to double and check bounds to prevent ensuing overflow.
       const double secondsD = static_cast<double>(seconds);
 
       if (secondsD >= kMaxSecondsD) {
         result = Timestamp(kMaxSeconds, kMaxNanoseconds);
-        return;
-      }
-      if (secondsD <= kMinSecondsD) {
+      } else if (secondsD <= kMinSecondsD) {
```

Comment:
Could we not do this refactor? This is not an essential refactor

Reply:
I’d like to keep this change to prevent returning true multiple times, which would be redundant.